docarray/array/storage/milvus/getsetdel.py

```diff
@@ -72,8 +72,7 @@ def _get_docs_by_ids(self, ids: 'Iterable[str]', **kwargs) -> 'DocumentArray':
                 raise KeyError(f'No documents found for ids {ids}')
             docs.extend(self._docs_from_query_response(res))
         # sort output docs according to input id sorting
-        id_to_index = {id_: i for i, id_ in enumerate(ids)}
-        return DocumentArray(sorted(docs, key=lambda d: id_to_index[d.id]))
+        return DocumentArray([docs[d] for d in ids])
```

Comment:
Can we keep the dict-based approach, but have every `id_` point to a list of positions, and leverage that in `key=...` somehow?
The reason it was done this way is performance, we want to gather the id-to-position mapping only once, and then delegate everything else to `sorted()`, which leverages a fist implementation in C.

Reply:
I can't think of a python built-in function. `sort` doesn't make the list longer than the original one.

Reply:
Ok I see, then let's keep it.